g4f/Provider/HuggingChat.py

```diff
@@ -101,24 +101,37 @@ async def create_async_generator(
             }
             options_json=json.dumps(options)
             body = (
-                    '------Web\r\n'
+                    '------WebKitFormBoundary*\r\n'
                     'Content-Disposition: form-data; name="data"\r\n\r\n'
                     f'{options_json}\r\n'
-                    '------Web--'
+                    '------WebKitFormBoundary*--'
             )
 
-            async with session.post(f"{cls.url}/conversation/{conversation_id}", headers=headers, data=body) as response:
-                async for line in response.content:
-                    await raise_for_status(response)
-                    line = json.loads(line)
-                    if "type" not in line:
-                        raise RuntimeError(f"Response: {line}")
-                    elif line["type"] == "stream":
-                        token = line["token"]
-                        token = token.lstrip().replace('\u0000', ' ')
-                        yield token
-                    elif line["type"] == "finalAnswer":
-                        break
+            async with session.post(f"{cls.url}/conversation/{conversation_id}", headers=headers, data=body) as response:  
+                await raise_for_status(response)
+                if stream:
+                    async for line in response.content:
+                        line = json.loads(line)
+                        if "type" not in line:
+                            raise RuntimeError(f"Response: {line}")
+                        elif line["type"] == "stream":
+                            token = line["token"].replace('\u0000', '')
+                            yield token
+                        elif line["type"] == "finalAnswer":
+                            break
+                else:
+                    content = await response.read()
```

Comment:
Added new stream=False feature, where it returns the final whole message after getting all the data from streamreader via await response.read()